src/Workspaces/Core/Portable/Extensions/OperationExtensions.cs

```diff
@@ -130,6 +130,32 @@ public static ValueUsageInfo GetValueUsageInfo(this IOperation operation)
                         return ValueUsageInfo.Read;
                 }
             }
+            else if (operation.Parent is IReturnOperation returnOperation)
+            {
+                switch (returnOperation.RefKind)
+                {
+                    case RefKind.RefReadOnly:
+                        return ValueUsageInfo.ReadableReference;
+
+                    case RefKind.Ref:
+                        return ValueUsageInfo.ReadableWritableReference;
+
+                    default:
+                        return ValueUsageInfo.Read;
+                }
+            }
+            else if (operation.Parent is IConditionalOperation conditionalOperation)
+            {
+                if (operation == conditionalOperation.WhenTrue
+                    || operation == conditionalOperation.WhenFalse)
+                {
+                    return GetValueUsageInfo(conditionalOperation);
```

Comment:
is this true? would it depend on how the conditional was used *and* if 'ref' was used inside the conditional?  (i'm honeslty not sure).

Reply:
The return type of the method determines the behavior here.

* If the method returns by-value or by-ref-readonly, the branches of the conditional will be treated as readonly references
* If the method returns by-ref, the branches of the conditional will be `ref` and failure to include the `ref` around the whole conditional would be a compilation error